signer/src/transaction_signer.rs

```diff
@@ -758,6 +708,24 @@ where
         Ok(())
     }
 
+    /// Check whether we will sign the message, which is supposed to be
+    /// bitcoin sighash
+    async fn validate_bitcoin_sign_request<D>(db: &D, message: &[u8]) -> Result<(), Error>
+    where
+        D: DbRead,
+    {
+        let sighash = TapSighash::from_slice(message)
+            .map_err(Error::SigHashConversion)?
+            .into();
+
+        if db.will_sign_bitcoin_tx_sighash(&sighash).await? != Some(true) {
+            tracing::warn!(%sighash, "sighash unknown or invalid");
+            return Err(Error::InvalidSigHash);
```

Comment:
could it be useful to differentiate, just in the term of logs, between an unknown sighash and one that exists in the database but was intentionally decided we will not sign? i.e. a sighash could be "valid," but we may have decided not to sign it because of our block list.

Reply:
Fixed in https://github.com/stacks-network/sbtc/pull/930/commits/1187594e2e9a9b4c3ccb6e01ea045e6f0af34a17